### PR TITLE
561449/lic emf

### PR DESCRIPTION
**Pull request by ruspl-afed**
Add an API to retrieve EMF metadata for java type

### CODE REVIEW

File: bundles/org.eclipse.passage.lic.emf/src/org/eclipse/passage/lic/emf/meta/ClassMetadata.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0/.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     ArSysOp - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.passage.lic.emf.meta;
```
Comment (eparovyshnaya) [inline bundles/org.eclipse.passage.lic.emf/src/org/eclipse/passage/lic/emf/meta/ClassMetadata.java]: Can we start as `internal`, please?

File: bundles/org.eclipse.passage.lic.emf/src/org/eclipse/passage/lic/emf/meta/ComposedClassMetadata.java [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+	public ComposedClassMetadata() {
+		registry = new LinkedHashSet<>();
+	}
+
+	@Override
+	public Optional<EntityMetadata> find(Class<?> clazz) {
+		return registry.stream()//
+				.map(m -> m.find(clazz))//
+				.filter(o -> o.isPresent())//
```
Comment (eparovyshnaya) [inline bundles/org.eclipse.passage.lic.emf/src/org/eclipse/passage/lic/emf/meta/ComposedClassMetadata.java]:
```java
.filter(Optional::isPresent) 
```
is preferrable

File: bundles/org.eclipse.passage.lic.emf/src/org/eclipse/passage/lic/emf/meta/ComposedClassMetadata.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+				.map(o -> o.get())//
```
Comment (eparovyshnaya) [inline bundles/org.eclipse.passage.lic.emf/src/org/eclipse/passage/lic/emf/meta/ComposedClassMetadata.java]:
```java
.map(Optional::get)
```

File: tests/org.eclipse.passage.lic.emf.tests/src/org/eclipse/passage/lic/emf/meta/ComposedClassMetadataTest.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+public class ComposedClassMetadataTest {
+
+	private final EClass type = EcoreFactory.eINSTANCE.createEClass();
+	private final EStructuralFeature id = EcoreFactory.eINSTANCE.createEAttribute();
+	private final EStructuralFeature name = EcoreFactory.eINSTANCE.createEReference();
+	private final EntityMetadata metadata = new PlainEntityMetadata(type, id, name);
+	private final ClassMetadata searcher = c -> Optional.of(metadata);
+
+	@Test(expected = NullPointerException.class)
+	public void testNullConsider() {
```
Comment (eparovyshnaya) [inline tests/org.eclipse.passage.lic.emf.tests/src/org/eclipse/passage/lic/emf/meta/ComposedClassMetadataTest.java]: `test` prefix for `@Test`-method is no longer mandatory, we can name tests in a less verbose way with more sence.

File: bundles/org.eclipse.passage.lic.emf/src/org/eclipse/passage/lic/emf/meta/ClassMetadata.java (hunk 1 above)
Comment (ruspl-afed) [reply bundles/org.eclipse.passage.lic.emf/src/org/eclipse/passage/lic/emf/meta/ClassMetadata.java]: Done

File: bundles/org.eclipse.passage.lic.emf/src/org/eclipse/passage/lic/emf/meta/ComposedClassMetadata.java (hunk 2 above)
Comment (ruspl-afed) [reply bundles/org.eclipse.passage.lic.emf/src/org/eclipse/passage/lic/emf/meta/ComposedClassMetadata.java]: cool

File: bundles/org.eclipse.passage.lic.emf/src/org/eclipse/passage/lic/emf/meta/ComposedClassMetadata.java (hunk 3 above)
Comment (ruspl-afed) [reply bundles/org.eclipse.passage.lic.emf/src/org/eclipse/passage/lic/emf/meta/ComposedClassMetadata.java]: cool

File: tests/org.eclipse.passage.lic.emf.tests/src/org/eclipse/passage/lic/emf/meta/ComposedClassMetadataTest.java (hunk 4 above)
Comment (ruspl-afed) [reply tests/org.eclipse.passage.lic.emf.tests/src/org/eclipse/passage/lic/emf/meta/ComposedClassMetadataTest.java]: wow! I'll consider it next time

**Pull request merged**